pkg/store/bucket.go

```diff
@@ -456,6 +461,16 @@ func (s *BucketStore) addBlock(ctx context.Context, id ulid.ULID) (err error) {
 	lset := labels.FromMap(b.meta.Thanos.Labels)
 	h := lset.Hash()
 
+	// Check for block labels by relabeling.
+	// If output is empty, the block will be dropped.
+	processedLabels := relabel.Process(promlabels.FromMap(lset.Map()), s.relabelConfig...)
+	if processedLabels == nil {
+		level.Debug(s.logger).Log("msg", "dropping block(drop in relabeling)", "block", id)
+		return os.RemoveAll(dir)
+	}
+	b.labels = labels.FromMap(processedLabels.Map())
```

Comment:
This is important because actually we get the labels after relabelling, which we agreed not doing in the first iteration (:

Reply:
So should we expose the original labels from `meta.json`? Maybe the action of relabeling will be `labeldrop`.

Reply:
Yes!

It may yes. But what's the use case to drop only some labels?

In fact it would be more useful to add new labels! This is great but there are some limitiation for it (e.g relabelling cannot generate 2 lablesets) . For now in this PR I would say we care ONLY if it exists or not exists. Does it sounds fair? (: